trolo/export/exporter.py

```diff
@@ -0,0 +1,154 @@
+        # Load checkpoint
+        checkpoint = torch.load(model, map_location="cpu", weights_only=False)
+
+        if "HGNetv2" in self.config.yaml_cfg:
+            self.config.yaml_cfg["HGNetv2"]["pretrained"] = False
+
+        # Load model state
+        if "ema" in checkpoint:
+            state = checkpoint["ema"]["module"]
+        else:
+            state = checkpoint["model"]
+
+        # Load state into config.model
+        self.config.model.load_state_dict(state)
+
+        # Create deployment model wrapper
+        model = self.config.model.deploy()
+
+        return model
+
+    def export(
+        self, 
+        export_format ,
+        input_size ,
+        input_data , 
+        input_names  , 
+        output_names ,
+        dynamic_axes,
+        batch_size : int =  1,
+        opset_version : int = 16,
+        simplify : bool = False,
+    ):
+        # check the model format
+        if export_format is None:
+            raise ValueError(f"Export format is missing!")
+        
+        #  check for export format
+        accepted = self._filter_format(export_format)
+
+        if export_format.lower().strip() == "onnx":
+            self._export2onnx(
+                input_size=input_size, 
+                input_data = input_data, 
+                input_names =  input_names, 
+                output_names = output_names,
+                dynamic_axes= dynamic_axes,
+                batch_size = batch_size, 
+                simplify = simplify, 
+                opset_version = opset_version
+            )
+
+
+    def _export2onnx(
+        self,
+        input_size : Union[int, List, Tuple, torch.Tensor] = None,
+        input_data : Union[int, List, Tuple, torch.Tensor] = None, 
+        input_names : Optional[list] =  None, 
+        output_names : Optional[list] =  None,
+        dynamic_axes : Optional [dict] =  None,
+        batch_size : int =  1,
+        opset_version : int = 16,
+        simplify : bool = False,
+        export_format : str = "onnx"
+    ) -> None: 
+        input_size  = format_size(input_size)
+        input_data  = format_input(input_data, batch_size=batch_size)
+        exported_path  =  get_output_name(model_path = self.model_path, export_format = export_format)
+
+        dynamic_axes = dynamic_axes  or {'images': {0: 'N', },'orig_target_sizes': {0: 'N'}}
+        torch.onnx.export(
+            self.model,
+            (input_data, input_size),
+            exported_path,
+            input_names = input_names, 
+            output_names = output_names,
+            dynamic_axes=dynamic_axes,
+            opset_version=16,
+            verbose=False,
+            do_constant_folding=True,
+        )
+
+        # TODO -  Exceptional Handler 
+        onnx_model  = onnx.load(exported_path)
+        onnx.checker.check_model(onnx_model)
+        LOGGER.info(f"Model exported to ONNX: {exported_path}")
+
+        if simplify:
+            self._simplify_onnx(exported_path)
+
```

Comment:
dry run on exported model is missing. It should be an optional with an argument.

Reply:
resolved.